internal/pkg/v2/infrastructure/redis/client.go

```diff
@@ -35,6 +35,8 @@ func NewClient(config db.Configuration, logger logger.LoggingClient) (*Client, e
 	if err != nil {
 		return nil, errors.NewCommonEdgeX(errors.KindDatabaseError, "redis client creation failed", err)
 	}
+	go dc.AsyncDeleteReadings()
+	go dc.AsyncDeleteEvents()
```

Comment:
This is a core data specific action, so it doesn't belong in the common NewClient code. i.e. What happens when Core Metadata calls this NewClient() func? 

Suggest these calls go in Code Data bootstrap handler once the client is created. These calls could go here:
https://github.com/edgexfoundry/edgex-go/blob/151af978cf25de30735135da97f91b1f15ae3e78/internal/core/data/init.go#L61
Since this handler runs after the V2 DBCleint has been created and is in the DIC.

Reply:
I moved the go routines start up to the very end of init.go.